modules/deeptools_bamFingerprint.cfmod.pl

```diff
@@ -0,0 +1,133 @@
+# Cluster Flow is free software: you can redistribute it and/or modify   #
+# it under the terms of the GNU General Public License as published by   #
+# the Free Software Foundation, either version 3 of the License, or      #
+# (at your option) any later version.                                    #
+#                                                                        #
+# Cluster Flow is distributed in the hope that it will be useful,        #
+# but WITHOUT ANY WARRANTY; without even the implied warranty of         #
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          #
+# GNU General Public License for more details.                           #
+#                                                                        #
+# You should have received a copy of the GNU General Public License      #
+# along with Cluster Flow.  If not, see <http://www.gnu.org/licenses/>.  #
+##########################################################################
+
+my %requirements = (
+        'cores'         => ['2', '4'],
+        'memory'        => ['4G', '8G'],
+        'modules'       => '',
+        'time'          => sub {
+
+                my $cf = $_[0];
+                my $num_files = $cf->{'num_starting_merged_aligned_files'};
+                # Default to 1 if none were parsed for whatever reason
+                $num_files = ($num_files > 0) ? $num_files : 1;
+                # Ususally takes around 5 mins per file. Set to 30 mins just to be sure.
+                return CF::Helpers::minutes_to_timestamp ($num_files * 1 * 30);
+        }
+);
+
+
+# The help text
+my $helptext = "".("-"x15)."\n DeepTools bamFingerprint\n".("-"x15)."\nTakes two input files: a bam file and a _crosscorrelation.txt file, as created by phantompeaktools. Creates a \"fingerprint\" plot of the read coverage, using deepTools/bamFingerprint. Outputs are are plot files named basename_fingerprint.png. The fragment length is set using 1) the input parameter fragmentLength 2) the file _crosscorrelation.txt and 3) default fragment length of 200 (if no input parameter is set and the file _crosscorrelation.txt doesn't exist). Currently expects a locally installed bamFingerprint.\n\n";
+
+
+# Start your engines...
+my %cf = CF::Helpers::module_start(\%requirements, $helptext);
+
+my $defaultFragLen = 200; # If fragment length not defined, use this value. 
+
+
+# Print version information about the program to be executed.
+warn "---------- bamFingerprint  version information ----------\n";
+warn `bamFingerprint --version`;
+warn "\n------- End of bamFingerprint version information ------\n";
+
+# Open up our run file in append mode
+open (RUN,'>>',$cf{'run_fn'}) or die "###CF Error: Can't write to $cf{run_fn}: $!";
+
+# Go through each file and run our command
+foreach my $file (@{$cf{'prev_job_files'}}){
+    # Start the clock...
+    my $timestart = time;
+
+    # Check that input is bed file
+    if(!($file =~ /\.bam$/)){
+       	next;
```

Comment:
Done! Fixed so it's case insensitive. I'm not sure there should be an error message here, because this module can take 2 input files from he previous module in the pipeline: a bam file, and an (optional) txt file with fragement length etc. (made by phantompeaktools). So it's not always weird to see a non bam file as input.